python/lsst/ts/ledprojector/led_object.py

```diff
@@ -0,0 +1,189 @@
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["LedObject"]
+import asyncio
+import inspect
+
+
+from lsst.ts import salobj, utils
+from lsst.ts.idl.enums.run_ledprojector import (
+    LEDBasicState,
+)
+from .labjack_interface import LabJackInterface
+from .led_object import LEDObject
+from .mock_labjack_interface import MockLabJackInterface
+
+# DRA-CN024D05 SSR requires 3.0VDC minimum turn-on voltage, up to 12 VDC max
+# 1.0VDC minimum turn-off voltage
+# all ports should be digital IO so write 1 for on/0 for off
+# Values to write to labjack for operating LED
+LED_OFF_SETTING = 0
+LED_ON_SETTING = 1
+
+class LEDObject:
+    """Class to capture the states and handling of the singular specific LEDs connected to a labjack
+
+    Parameters
+    ----------
+    config : `types.SimpleNamespace`
+        LED-specific configuration.
+    csc : `lsst.ts.salobj.BaseCsc`
+        The CSC. This class writes to led-specific event topics
+        and reads and writes two additional float attributes:
+        ``led_on_time`` and ``led_off_time``.
+    log : `logging.Logger`
+        Logger.
+    labjack : 'LabJackInterface'
+        The defined interface class to the labjack that this LED is connected to.
+    labjack_port : 'string'
+        The string name of the Labjack port that this LED is connected to.
+    led_name : 'string'
+        String of the serial number for this LED model.
+    status_callback : `awaitable` or `None`, optional
+        Coroutine to call when evt_ledState or evt_ledConnected changes.
+        It receives one argument: this model.
+    simulate : `bool`, optional
+        Run in simulation mode?
+    make_connect_time_out : `bool`, optional
+        Make the connect method timeout?
+        Only useful for unit tests.
+        Ignored if simulate false.
+
+    Raises
+    ------
+    TypeError
+        If ``status_callback`` is not None and not a coroutine.
+    ValueError
+        If ``config.default_power`` not in range [800, 1200], inclusive.
+
+    Attributes
+    ----------
+    default_power : `float`
+        Default power after the led is started (Watts) in range 800-1200.
+    led_was_on : `bool`
+        Was the led commanded on, as of the most recently read LabJack data?
+    """
+
+    def __init__(
+        self,
+        config,
+        csc,
+        log,
+        labjack,
+        labjack_port,
+        led_name,
+        status_callback=None,
+        simulate=False,
+        make_connect_time_out=False,
+    ):
+        if status_callback is not None and not inspect.iscoroutinefunction(
+            status_callback
+        ):
+            raise TypeError(
+                f"status_callback={status_callback} must be None or a coroutine"
+            )
+
+        self.labjack_port = labjack_port
+        self.led_name = led_name
+        self.config = config
+        self.csc = csc
+        self.log = log.getChild("LEDObject")
```

Comment:
Consider allowing `log` to be `None` in which case you create a new logger, e.g.;

```
self.log = log.getChild(type(self).__name__) if log is not None else logging.getLogger(type(self).__name__)
```